### PR TITLE
Call Next Media API for video data

### PR DESCRIPTION
**Pull request by benbarnett**
**Breaking change.**

* The component now calls the [next-media-api](https://github.com/Financial-Times/next-media-api), which supports both brightcoveIds and uuids
* Removes `captionsUrl` option, adds `showCaptions` – saves users having to make a separate call to lookup the captions by using the URL directly from the response data. Video data can still be passed in via data attribute to avoid the XHR if necessary.
* If `showCaptions` is `true`, calling `addVideo()` directly will throw an error. This is due to the fact the component needs to source the captionsUrl first.

### CODE REVIEW

File: test/video.test.js [hunk 1]
```diff
@@ -575,11 +602,15 @@ describe('Video', () => {
...
-			setTimeout(() => {
-				video.getTrackMode().should.equal(undefined);
-			}, 100);
+
+			return video.init().then(() => {
+				video.addVideo();
+				setTimeout(() => {
+					video.getTrackMode().should.equal(undefined);
+				}, 100);
+			});
```
Comment (JakeChampion) [inline test/video.test.js]: This will resolve before the timeout executes

Comment (benbarnett) [reply test/video.test.js]: Good point. Although I am struggling to recreate these failures locally 🤔

File: README.md [hunk 2]
```diff
@@ -27,8 +27,8 @@ const video = new OVideo(document.body, opts);
 
 Where `opts` is an optional object with properties
 
- * `id` [`Number`] Source's ID of the video
- * `source` [`String`] Source of the video (currently only accepts `brightcove`)
+ * `id` [`Number`] Source's ID of the video (`brightcoveId` or `uuid`)
```
Comment (rowanbeentje) [inline README.md]: No longer a `Number` :)

Comment (benbarnett) [reply README.md]: An excellent point!

File: README.md [hunk 3]
```diff
@@ -75,11 +75,32 @@ The queue is an `array` containing Brightcove video ID strings.
 ```
 $ npm test
 ```
-(Requires Firefox)
+Requires Firefox (v38.0.0 to test with polyfills and mirror CircleCI)
 
 
 ## Migration Guide
 
+Migrating from 2.0 to 3.0
```
Comment (rowanbeentje) [inline README.md:84]: Previously o-video could also be set up with a data object matching that fetched from next-video/api - worth explicitly mentioning that this should still work if it's set up with a different data object matching that fetched from next-media-api?

Comment (benbarnett) [reply README.md:84]: Good idea, will mention this

Comment (rowanbeentje) [reply README.md:84]: (Otherwise this seems to be working!)

**Comment by benbarnett** [comment]
@JakeChampion right, we're (finally) green now and should be ready to go

**Review by JakeChampion — APPROVED** [review]

**Pull request merged**